lib/compress.js

```diff
@@ -2582,6 +2598,18 @@ merge(Compressor.prototype, {
 
     OPT(AST_Call, function(self, compressor){
         var exp = self.expression;
+        if (compressor.option("reduce_vars")
+            && exp instanceof AST_SymbolRef) {
+            var def = exp.definition();
+            if (def.fixed) {
+                exp = def.fixed;
+                if (compressor.option("unused")
+                    && def.references.length == 1
+                    && exp instanceof AST_Function) {
```

Comment:
How were you able to get the test `defun_call` to work without using the following?
```js
                    && (exp instanceof AST_Function || exp instanceof AST_Defun)) {
```
Isn't `function g()` in the test an `AST_Defun`?